mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/SynchronousMavericksStateStore.kt

```diff
@@ -2,33 +2,26 @@ package com.airbnb.mvrx.mocking
 
 import androidx.annotation.RestrictTo
 import com.airbnb.mvrx.MavericksStateStore
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.channels.BroadcastChannel
-import kotlinx.coroutines.channels.Channel
-import kotlinx.coroutines.channels.consumeEach
-import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.channels.BufferOverflow
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.buffer
-import kotlinx.coroutines.flow.distinctUntilChanged
-import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
 
 /**
  * This acts as a functional state store, but all updates happen synchronously.
  * The intention of this is to allow state changes in tests to be tracked
  * synchronously.
  */
-@Suppress("EXPERIMENTAL_API_USAGE")
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-class SynchronousMavericksStateStore<S : Any>(initialState: S, coroutineScope: CoroutineScope) : MavericksStateStore<S> {
+class SynchronousMavericksStateStore<S : Any>(initialState: S) : MavericksStateStore<S> {
 
-    private val stateChannel = BroadcastChannel<S>(capacity = Channel.BUFFERED)
+    private val stateSharedFlow = MutableSharedFlow<S>(
+        replay = 1,
+        extraBufferCapacity = 64,
+        onBufferOverflow = BufferOverflow.SUSPEND,
+    ).apply { tryEmit(initialState) }
 
-    init {
-        coroutineScope.coroutineContext[Job]!!.invokeOnCompletion {
-            stateChannel.cancel()
-        }
-    }
+    override val flow: Flow<S> = stateSharedFlow.asSharedFlow()
```

Comment:
in this store it makes sense to add `.distinctUntilChanged()` because same values aren't dropped on reducer level. The same for Scriptable store